lib/ansiblelint/formatters/__init__.py

```diff
@@ -42,14 +42,15 @@ class Formatter(BaseFormatter):
 
     def format(self, match, colored=False):
         formatstr = u"{0} {1}\n{2}:{3}\n{4}\n"
+        _id = getattr(match.rule, 'id', '000')
```

Comment:
Maybe it makes sense to add a property to the rule base or match that would do this instead of us special-casing all the uses?

Reply:
I plan to rework this logic and introduce full typing checks, as these bugs would not have happened with them. Still, fixing 2.10 compatibility takes priority, so we should delay other refactorings until we pass tests with 2.10.